stan-ballerina/tests/nats_streaming_auth_tests.bal

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["nats-streaming"]
+}
+isolated function testConnectionWithToken() returns error?{
+    Tokens myToken = { token: "MyToken" };
+    boolean flag = false;
+    Client? con = check new("nats://localhost:4223", auth = myToken);
```

Comment:
Can remove `? ` from `Client?`

Reply:
For this test actually, you can remove the `check`. Keep the expected types as  `Client|Error` then check for the error

Reply:
Check other places also